dokan/examples/memfs/main.rs

```diff
@@ -499,14 +496,25 @@ impl<'c, 'h: 'c> FileSystemHandler<'c, 'h> for MemFsHandler {
 		let delete_on_close = create_options & FILE_DELETE_ON_CLOSE > 0;
 		let path_info = path::split_path(&self.root, file_name)?;
 		if let Some((name, parent)) = path_info {
+			if create_options & FILE_DIRECTORY_FILE > 0 {
+				if let Some(stream_info) = &name.stream_info {
+					if !stream_info.check_default(true)? {
```

Comment:
There is no other cleaner way to know if this is a file ?

Reply:
This is a check to see if an entry (file/directory) has streams. I think it was the code author's decision on how to handle streams.